app/src/androidTest/java/org/projectbuendia/client/ui/FunctionalTestCase.java

```diff
@@ -437,4 +447,37 @@ protected void inPatientListClickPatientWithId(String id) {
             .inAdapterView(hasId(R.id.fragment_patient_list))
             .atPosition(0));
     }
+
+    /** Generates IDs to identify the newly created patient. */
+    protected String generateId() {
+        String id = Long.toString(new Date().getTime()%100000);
+        return id;
+    }
+
+    /** Populates all the fields on the New Patient screen. */
+    protected void populateNewPatientFields(String id) {
+        screenshot("Before Patient Populated");
+        String given = "Given" + id;
+        String family = "Family" + id;
+        type(id, viewWithId(R.id.patient_id));
+        type(given, viewWithId(R.id.patient_given_name));
+        type(family, viewWithId(R.id.patient_family_name));
+        type(id.substring(id.length() - 2), viewWithId(R.id.patient_age_years));
+        type(id.substring(id.length() - 2), viewWithId(R.id.patient_age_months));
+        chooseSex();
+        screenshot("After Patient Populated");
+    }
+
+    /** Randomly choose a sex for the patient */
```

Comment:
This logic can be written more clearly as:

```
click(viewWithId(new Random().nextInt() % 2 == 0 ?
    R.id.patient_sex_female : R.id.patient_sex_male));
```

However, it's better to make the test more deterministic -- rather than generating a new random number, make the choice a function of the patient ID, e.g. `chooseSex(int id) { ... id % 2 == 0 ? R.id.patient_sex_female : R.id.patient_sex_male ... }`